### ISSUE TITLE
As an Administrator, I want to add front-end Users to my site

### ISSUE THREAD

**Issue by robyurkowski** [grounded/afterburnercms#26]
I want them to be able to log in without being able to see the back-end so that I can serve authenticated content to them and restrict content to non-authenticated users.

**Comment by GeekOnCoffee** [comment]
+1

**Comment by parndt** [comment]
Yes this is a must. We should also make the default authentication solution _OPTIONAL_.

**Comment by knewter** [comment]
So I _definitely_ think this needs to be a separate thing.  i.e. :+1: on
the optional bit.  Forem has done this correctly, and integrating forem
with other apps is dead simple.  I'd definitely prefer we _start out_
leaving this completely out of afterburner, except the 'launch a new app'
generator auto-generates a devise configuration to plug the mounting app's
users into afterburner.  Anything else is wrong and broken.

**Comment by robyurkowski** [comment]
That's the plan, yo. I'm looking forward to it.

**Comment by robyurkowski** [comment]
Highest Priority with a capital P.

@parndt, @knewter -- one of you chaps want to tackle this?

**Comment by knewter** [comment]
I've got a few minutes, thinking into it now.  I won't get finished before I have to grab a shower and head in to the office of the company I'm here visiting.  Let me lay out my plan in this comment first, that has value right?.
- First off, only address this one place at first (backend) but anticipate parts/all of it will be pulled into `abc-rails-common` eventually
- OK, so since it's largely a backend task at first (administer Users) then we have to provide some sort of form.
- The User model (or whatever it's called) will live in the hosting app.  Ripping this conceptually from Forem gives us:

``` ruby
# config/initializers/abc.rb
Abc.user_class = User
```

``` ruby
class ApplicationController
  def abc_user
    current_user
  end
  helper_method :abc_user
end
```
- That will get us the ability to talk about the hosting app's Users, but since we also intend to administer them (initially?  In some cases this may not be what people want really) we'll need to provide them the ability to override the form.  We can ship with a form that totes works with Devise users, and anticipate ripping that part out into a plugin (a default plugin installed by the installer?).  This is suboptimal, but at least by ripping out the concept of userhood from the get go we'll be in a spot where this is easier.
- Nothing here describes what it means to be an abc administrator.  I started to add an `abc_admin?` method in that controller but it seems foolish - first off, doubt Abc should by default derive from the main app's `ApplicationController`, and secondly we likely want `Abc::Entity::Role`s that can be layered upon arbitrary users.
- So `abc-core` needs to be where the `Role` (and some wrapper User entity concept?) lives, since Userhood is by its very nature part of the app I would assume.  However, obv. all this configuration goodness lives in the rails side of things.

Thoughts?  Things I left out or overthought?
